src/Bicep.Core/PrettyPrint/DocumentBuildVisitor.cs

```diff
@@ -0,0 +1,366 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Bicep.Core.Extensions;
+using Bicep.Core.Parser;
+using Bicep.Core.PrettyPrint.Documents;
+using Bicep.Core.PrettyPrint.DocumentCombinators;
+using Bicep.Core.Syntax;
+
+namespace Bicep.Core.PrettyPrint
+{
+    public class DocumentBuildVisitor : SyntaxVisitor
+    {
+        private static readonly ILinkedDocument Nil = new NilDocument();
+
+        private static readonly ILinkedDocument Space = new TextDocument(" ", Nil);
+
+        private static readonly ILinkedDocument Line = new NestDocument(0, Nil);
+
+        private static readonly IReadOnlyDictionary<string, TextDocument> CommonTextCache;
+
+        private readonly Stack<ILinkedDocument> documents = new Stack<ILinkedDocument>();
+
+        private readonly Stack<DocumentBlockContext> documentBlockContexts = new Stack<DocumentBlockContext>();
+
+        private readonly List<ILinkedDocument> precedingTrailingComments = new List<ILinkedDocument>();
+
+        static DocumentBuildVisitor()

```

Comment:
>static [](start = 8, length = 6)

nit - Could avoid the static constructor by either having a static method used to populate the dictionary or just by using linq (it seems like all we're doing is a bunch of selects that are concat'd together)

Reply:
Good catch! Replaced it with linq.